From b5b1be8a7bdc212cb2c21d9bff1148533b12057f Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Fri, 27 May 2022 15:17:34 +0300
Subject: [PATCH 1/8] CMake: Bump mininum version requirement to 3.5

Silences spam such as the following during CMake configure:

  CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.

Even xenial has cmake 3.5.1: https://launchpad.net/ubuntu/xenial/+source/cmake
---
 CMakeLists.txt                 | 2 +-
 cmake/UseConstantBuilder.cmake | 2 +-
 cmake/UseGdbusCodegen.cmake    | 2 +-
 cmake/UseGlibGeneration.cmake  | 2 +-
 cmake/UseLttngGenTp.cmake      | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4064afb..8c9af9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(lomiri-app-launch C CXX)
-cmake_minimum_required(VERSION 2.8.9)
+cmake_minimum_required(VERSION 3.5)
 
 ##########################
 # Version Info
diff --git a/cmake/UseConstantBuilder.cmake b/cmake/UseConstantBuilder.cmake
index 06c8d05..040432b 100644
--- a/cmake/UseConstantBuilder.cmake
+++ b/cmake/UseConstantBuilder.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
diff --git a/cmake/UseGdbusCodegen.cmake b/cmake/UseGdbusCodegen.cmake
index ddb2995..8419e7a 100644
--- a/cmake/UseGdbusCodegen.cmake
+++ b/cmake/UseGdbusCodegen.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
diff --git a/cmake/UseGlibGeneration.cmake b/cmake/UseGlibGeneration.cmake
index 67e34b9..e937b1c 100644
--- a/cmake/UseGlibGeneration.cmake
+++ b/cmake/UseGlibGeneration.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
diff --git a/cmake/UseLttngGenTp.cmake b/cmake/UseLttngGenTp.cmake
index 62c1ca8..45e6c45 100644
--- a/cmake/UseLttngGenTp.cmake
+++ b/cmake/UseLttngGenTp.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
-- 
2.36.1

